keyrings/codeartifact.py

```diff
@@ -193,3 +187,30 @@ def set_password(self, service, username, password):
     def delete_password(self, service, username):
         # Defer deleting a password to the next backend
         raise NotImplementedError()
+
+    @staticmethod
+    def get_boto_session(*, region, config):
+        should_assume_role = config.get("assume_role")
+
+        # Create session with any supplied configuration.
+        session = boto3.Session(
+            region_name=region,
+            profile_name=config.get("profile_name"),
+            aws_access_key_id=config.get("aws_access_key_id"),
+            aws_secret_access_key=config.get("aws_secret_access_key"),
+        )
+
+        if should_assume_role is not None:
+            assumed_role = session.client("sts").assume_role(
+                RoleArn=config["assume_role"],
+                RoleSessionName=config.get(
+                    "assume_role_session_name", "KeyRingsCodeArtifact"
```

Comment:
So the AWS SDK will not pick a `RoleSessionName` for you; you can see the [docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts/client/assume_role.html) that they say they are required. So I have include an explicit default as a fallback instead.

Reply:
As mentioned in a comment, I'd like this fallback to be a dynamic value usable for auditing in CloudTrail, but that's not required for this PR.